weave/trace/serialization/op_type.py

```diff
@@ -466,6 +471,15 @@ def _get_code_deps(
     return {"import_code": import_code, "code": code, "warnings": warnings}
 
 
+def _has_memory_address(obj: Any) -> bool:
+    return isinstance(obj, str) and MEMORY_ADDRESS_PATTERN.search(obj) is not None
+
+
+def _replace_memory_address(json_val: str) -> str:
+    """Turn <Function object at 0x10c349010> into <Function object at 0x000000000>"""
+    return MEMORY_ADDRESS_PATTERN.sub("0x000000000>", json_val)
```

Comment:
Optional - I think you could use a lambda for the replacement to ensure the replacement address was the same length as the input one.

Reply:
oh, yeah that's a nice idea thx.